misc/AL2023-Dockerfile

```diff
@@ -19,6 +19,12 @@ RUN dnf install -y \
     automake \
     autoconf
 
+# Some basic tools that are typical during builds
+RUN dnf install -y --allowerasing \
+    gnupg2 \
+    gcc \
+    make
```

Comment:
Note: I'm surprised that `make` needs to be explicit. I thought that when we install e.g. `automake` above, it automatically installs `make`. Are you sure this explicit `make` is needed?

On the other hand, it's better to be explicit, so I don't mind this change.